packages/inventory/src/ducks/inventory.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  fetchActionCreatorFactory,
+  removeActionCreatorFactory,
+  setTotalRecordsFactory,
+  reducerFactory,
+  getItemsArrayFactory,
+  getTotalRecordsFactory,
+} from '@opensrp/reducer-factory';
+import { values } from 'lodash';
+import { Store } from 'redux';
+import { createSelector } from 'reselect';
+import { Dictionary } from '@onaio/utils';
+
+/** interface inventory */
+export interface Inventory {
+  deliveryDate: string;
+  donor: string;
+  poNumber: number;
+  productName: string;
+  providerId: string;
+  quantity: number;
+  serialNumber: string;
+  servicePointId: string;
+  stockId: string;
+  unicefSection: string;
+}
+
+/** reducer name */
+export const inventoryReducerName = 'inventory';
+
+/** Item Reducer */
+export const inventoryReducer = reducerFactory<Inventory>(inventoryReducerName);
+
+// action
+/** actionCreator returns action to to add Item records to store */
+export const fetchInventories = fetchActionCreatorFactory<Inventory>(
+  inventoryReducerName,
+  'servicePointId'
+);
+export const removeInventories = removeActionCreatorFactory(inventoryReducerName);
+export const setTotalInventories = setTotalRecordsFactory(inventoryReducerName);
+
+// selectors
+export const getInventoriesArray = getItemsArrayFactory<Inventory>(inventoryReducerName);
+export const getTotalInventories = getTotalRecordsFactory(inventoryReducerName);
+
+export const getStore = (store: Partial<Store>) => {
+  return (store as Dictionary)[inventoryReducerName].objectsById as Dictionary<Inventory>;
+};
+
+interface Filters {
+  servicePointIds?: string[];
+  servicePointId?: string;
```

Comment:
we do not need the second, since if you want inventories for one single id you can use the first but give it an array of only one id, makes sense?